org.eclipse.lyo.oslc4j.codegenerator/src/org/eclipse/lyo/oslc4j/codegenerator/files/generateAdaptorServiceProviderFactory.mtl

```diff
@@ -94,8 +80,29 @@ public class [javaClassNameForFactory(aServiceProvider) /]
         super();
     }
 
-    public static ServiceProvider createServiceProvider(final String baseURI, final String title, final String description, final Publisher publisher, final Map<String,Object> parameterValueMap)
```

Comment:
This is conceptually wrong, since the generator actually create a separate SPFactory for each SP. so, the method is enough to call createServiceProvider. You don't need to import the SP classes in the for loop on line 61 above.

Suggestion:
1. Let's review and approve https://github.com/eclipse/lyo.designer/pull/88
2. Tell me what more you want to do (just move some setters (like setAbout() methods) to this factory class?)
3. I fix :-)